### ISSUE TITLE
Feature: Code Diff Support

### ISSUE THREAD

**Issue by heychazza** [atomiks/rehype-pretty-code#118]
Hi,

It would be great to have code diff support to the highlighting system, e.g. being able to specify `{-1-4,+5-10}` which would highlight the first few rows with a red `-` and background, with a green `+` for the others.

```diff
+ like
- this
```

But spans the entire code line

**Comment by IanMitchell** [comment]
I'd also like something like this! I took a look at the repo to see how doable implementing it myself would be, and it looks like the proposed syntax might risk colliding with the current implementation. This library parses anything in `{}` blocks and passes it to `parse-numeric-range`, which allows for negative numbers. I'm not clear on if this would actually trigger anything (I believe it checks to see if a line number is in the range, so I suspect it wouldn't) but I think a safer syntax might be:

```
```jsx +{1-2} -{3-4}
import { thing } from "lib";
thing();
const thing = require("lib");
thing();
```

I might give implementing this a shot; if the maintainers are ok with a feature for this, changing the syntax based on feedback shouldn't be too difficult if I'm able to make it work!

**Comment by atomiks** [comment]
Fixed with `v0.12.0`, you can use [`shikiji-transformers`](https://github.com/antfu/shikiji/tree/main/packages/shikiji-transformers#transformernotationdiff)

<img width="894" alt="Screenshot 2023-12-03 at 6 22 42 pm" src="https://github.com/atomiks/rehype-pretty-code/assets/22450188/c64cdf7f-889a-432f-bbd1-f80926284c9f">
